src/components/POLine/hooks/useLocationsByHoldingIds/useLocationsByHoldingIds.js

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from 'react-query';
+
+import {
+  useOkapiKy,
+  useNamespace,
+} from '@folio/stripes/core';
+import { batchRequest } from '@folio/stripes-acq-components';
+
+import { HOLDINGS_API } from '../../../Utils/api';
+
+const DEFAULT_VALUE = [];
+
+export const useLocationsByHoldingIds = (holdingIds = []) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'holdings-by-id' });
+
+  const fetchFn = ({ params: searchParams }) => (
+    ky.get(HOLDINGS_API, { searchParams })
+      .json()
+      .then(({ holdingsRecords }) => holdingsRecords)
+  );
+
+  const {
+    data = {},
+    isLoading = false,
+    isFetching = false,
+  } = useQuery(
+    [namespace, holdingIds],
+    async () => {
+      const holdings = await batchRequest(
+        fetchFn,
+        holdingIds,
+      );
+
+      const permanentLocationIds = holdings.map(({ permanentLocationId }) => permanentLocationId);
+
+      return {
+        permanentLocationIds,
```

Comment:
Sorry, not quite understood it. What is your suggestion?

Reply:
It looks like I didn't describe the previous comment quite correctly. I suggest having a hook `useHoldingsByIds` which returns a list of holdings instead of a list of its permanent locations IDs. And map permanent location IDs on purpose in components (e.g. [here](https://github.com/folio-org/ui-orders/pull/1543/files/32a702706b0435a9c304a80eb761351f0254f067#diff-9648e7051d8d28c80e88147d177c2b4a5ef5b97b37c5339a45b5752148ace607R24)). This hook is easier to reuse and it can be stored in `common/hooks`.